### DISCUSSION TITLE
When clicking "x" in search bar, "q" parameter clears but "f" parameter remains. How to clear the "f"parameter as well?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#3151] [Q&A]
Hi. I have a hard coded link that filters a refinablestring via "category".
It works as it should .
However when a user clears the searchbox with "x"  in the searchbox ,or types in a new word, the "q" value is removed , but the "f" parameter and its value remains in the URL. 
How do I clear the "f' value as well.
I think I saw a reset search on PnP Search v3 but I do not see one in PnP v4.
Thank you for any assistance.

In PnP Search Results:
- I have input query text toggled on
- Dynamic value
- Page environment
- Query String
- Query Parameters
- queryParameter's propertis - q

**Comment by wobba** [comment]
Duplicate of #3153

**Reply by tedtninety** [reply]
Please accept my apologies.  It was an earnest forgetting.

**Reply by wobba** [reply]
np 🙂
